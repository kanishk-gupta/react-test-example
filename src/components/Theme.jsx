import React, { useState } from "react";

const Theme = () => {
  const [theme, setTheme] = useState("light");

  const toggleTheme = () => {
    setTheme(theme => (theme === "light" ? "dark" : "light"));
  };

  return (
    <button onClick={toggleTheme}>
      Current theme: {theme}
    </button>
  );
}

export default Theme;
