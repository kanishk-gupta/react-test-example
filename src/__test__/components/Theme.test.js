import { render, screen, cleanup } from '@testing-library/react';
import userEvent from "@testing-library/user-event";

import Theme from '../../components/Theme';

afterEach(cleanup);


describe('Theme test cases', () => {

  test('Theme', async () => {
    // ARRANGE
    render(<Theme />);

    // ACT
    const buttonEl = screen.getByText(/Current theme/i);
    await userEvent.click(buttonEl);

    // ASSERT
    expect(buttonEl).toHaveTextContent(/dark/i);
  });

  it('should take a snapshot', () => {
    const { asFragment } = render(<Theme />)
    
    expect(asFragment(<Theme />)).toMatchSnapshot();
  });

});
