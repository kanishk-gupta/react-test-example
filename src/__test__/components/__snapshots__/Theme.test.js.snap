// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme test cases should take a snapshot 1`] = `
<DocumentFragment>
  <button>
    Current theme: light
  </button>
</DocumentFragment>
`;
